packages/ui/src/primitives/separator.tsx

```diff
@@ -1,19 +1,38 @@
-import { cx } from 'cva';
 import * as React from 'react';
 
-export const Separator = React.forwardRef<HTMLParagraphElement, React.HTMLAttributes<HTMLParagraphElement>>(
-  function Separator({ children, className, ...props }, forwardedRef) {
-    return (
-      <p
-        ref={forwardedRef}
-        {...props}
-        className={cx(
-          'before:bg-gray-a4 after:bg-gray-a4 text-gray-a11 flex items-center gap-x-4 text-base before:h-px before:flex-1 after:h-px after:flex-1',
-          className,
-        )}
-      >
-        {children}
-      </p>
-    );
+import {
+  mergeDescriptors,
+  type ParsedDescriptor,
+  type ParsedElementsFragment,
+  useAppearance,
+} from '~/contexts/AppearanceContext';
+
+export const layoutStyle = {
+  separator: {
+    className: 'flex items-center gap-x-4 before:h-px before:flex-1 after:h-px after:flex-1',
+  },
+} satisfies ParsedElementsFragment;
+
+export const visualStyle = {
+  separator: {
+    className: 'before:bg-gray-a4 after:bg-gray-a4 text-gray-a11 text-base ',
   },
-);
+} satisfies ParsedElementsFragment;
+
+export const Separator = React.forwardRef<
+  HTMLParagraphElement,
+  React.HTMLAttributes<HTMLParagraphElement> & {
+    descriptors?: ParsedDescriptor[];
+  }
+>(function Separator({ children, descriptors, ...props }, forwardedRef) {
+  const { elements } = useAppearance().parsedAppearance;
+  return (
+    <p
```

Comment:
Curious if this is best as a `p` tag. Feels like it should be a `div`, similar to how Radix handles it.

Reply:
Good question, to my knowledge, that'd be for horizontal rule type elements that do not contain text. If I add `role="separator"` to this element all that gets read to screen readers is `horizontal splitter` and `or` is completely ignored.

![Screenshot 2024-09-16 at 1 05 51 PM](https://github.com/user-attachments/assets/e22a8876-95c0-46a9-8695-df48d84f4821)

Reply:
Similar RAC implementation. Neither show examples of text used in these components https://react-spectrum.adobe.com/react-spectrum/Divider.html